src/app/api/boosts/route.ts

```diff
@@ -0,0 +1,132 @@
+  cta_url: string | null;
+  cta_text: string | null;
+  boost_type: string;
+  is_enabled: boolean;
+  points: bigint;
+  nft_amount: bigint | null;
+};
+
+export type Boost = {
+  id: bigint;
+  name: string;
+  contractAddresses: string[];
+  imageUrl: string | null;
+  gameId: bigint;
+  ctaUrl: string | null;
+  ctaText: string | null;
+  boostType: string;
+  isEnabled: boolean;
+  points: bigint;
+  nftAmount: bigint | null;
+};
+
+async function mapBoostState(
+  boost: BoostType
+): Promise<Boost> {
+  return {
+    id: boost.id as bigint,
+    name: boost.name,
+    contractAddresses: boost.contract_addresses,
+    imageUrl: boost.image_url,
+    gameId: boost.game_id as bigint,
+    ctaUrl: boost.cta_url,
+    ctaText: boost.cta_text,
+    boostType: boost.boost_type,
+    isEnabled: boost.is_enabled,
+    points: boost.points,
+    nftAmount: boost.nft_amount
+  };
+}
+
+export async function GET(request: NextRequest) {
+  const searchParams = request.nextUrl.searchParams;
+  const userAddress = searchParams.get('userAddress') as string;
+  const gameId = BigInt(searchParams.get('gameId') as string);
+
+  if (!userAddress || !gameId) {
+    return new Response(
+      `Missing parameters: userAddress: ${userAddress}, gameId: ${gameId}`,
+      {
+        status: 400,
+      }
+    );
+  }
+
+  const boostsData = await supabase
+  .from('boost_configuration')
+  .select(`
+    *,
+    claimed_boost (
+      boost_id,
+      contract_address,
+      updated_at
+    )
+  `)
+  .eq('game_id', gameId)
+  .eq('is_enabled', true)
+  .filter('claimed_boost.user_address', 'eq', userAddress);
+
+  if (boostsData.error) {
+    console.error(boostsData.error);
+    throw new Error(boostsData.error.message);
+  }
+
+  const boosts = boostsData.data;
+
+  const boostsFormatted = boosts.map(async boost => {
+    let isClaimed = false;
+    const claimedAddresses = boost.claimed_boost.filter(b => b.contract_address).map(b => b.contract_address);
+    switch (boost.boost_type) {
+        case 'NFT':
+        case 'NFT_PER_MINT':
+            isClaimed = boost.contract_addresses.every(a => claimedAddresses.includes(a));
+            break;
+        case 'TOKEN':
+            isClaimed = boost.claimed_boost.every(c => c.boost_id === boost.id);
+            break;
+        case 'RECURRING':
+            const activeClaims = boost.claimed_boost.filter(c => {
+                const refreshHours = boost.refresh_time ? boost.refresh_time.getHours() : 24;
+                const expirationTime = new Date(c.updated_at.getTime() + hoursToMilliseconds(refreshHours));
+                return new Date() < expirationTime;
+            });
+            isClaimed = activeClaims.length > 0;
+            break;
+    }
+    isClaimed = isClaimed && boost.claimed_boost.length > 0;
+
+    const mappedBoost = await mapBoostState(boost);
+    return { ...mappedBoost, claimed: isClaimed };
+  });
+
+  return Promise.all(boostsFormatted).then(formattedBoosts => {
```

Comment:
shouldn't `boostsFormatted` already resolved to an array since you are awaiting on line 114?